FlowCrypt/Controllers/Msg/MessageActionsHandler.swift

```diff
@@ -30,10 +30,17 @@ extension MessageActionsHandler where Self: UIViewController {
     }
 
     func setupNavigationBar() {
-        trashFolderProvider.getTrashFolderPath()
-            .then(on: .main) { [weak self] path in
-                self?.setupNavigationBarItems(with: path)
+        Task {
+            do {
+                let path = try await trashFolderProvider.getTrashFolderPath()
+                DispatchQueue.main.async {
```

Comment:
Better to avoid having to different concurrency approaches in one place

Reply:
Agree - is it the same issue as #997 ?

Reply:
Yes